### ISSUE TITLE
Parsing a pkcs#12 file

### ISSUE THREAD

**Issue by vinaykarthikr** [ARMmbed/mbedtls#4749]
Hi, 
I am in a project where we get a pkcs12(pfx) file and send it to our device where the file is parsed and the private key and 
the certificate are stored in separate files for use in TLS communications. As far as i can see there is a function called mbedtls_pkcs12_pbe which looks like the function that should serve my need but it requires cipher_type and md_type which we dont know(can be anything based on the pfx generation) and can know only after successful parsing of the pkcs12 file.

### Enhancement

It would be good if there was a function which parsed the pfx file with the password and populated the result into separate buffers for key and certificate contained in the file.

### Justification

Mbed TLS needs this because pkcs#12/pfx is a very popular extension used in windows and this would allow the users to easily import and use pfx files.

**Comment by ronald-cron-arm** [comment]
The "PKCS #12: Personal Information Exchange Syntax v1.1" specification is now the RFC 7292. If I understand correctly the request for enhancement is to determine from the parsing of the PKCS12 file the cipher and message digest used for encryption and authentication rather than relying on this to be determined beforehand.

**Comment by vinaykarthikr** [comment]
Yes, your understanding is correct.  The only detail that we have other than the PKCS12 file itself is the password, if one was used for creating it.
